downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/FetchTestReactomeOntologyFiles.java

```diff
@@ -0,0 +1,137 @@
+		PrintWriter pontWriter = new PrintWriter(pontFilename);
+		PrintWriter pinsWriter = new PrintWriter(pinsFilename);
+		
+		while (resultSet.next()) {
+			
+			Blob blob = resultSet.getBlob("ontology");
+			BufferedReader br = new BufferedReader(new InputStreamReader(blob.getBinaryStream()));
+
+			int dateTimeCounter = 0;
+			boolean pprjSwitch = true;
+			boolean pontSwitch = false;
+			String str;
+			logger.info("Generating " + pprjFilename + ", " + pontFilename + ", and " + pinsFilename);
+			while ((str = br.readLine()) != null) {
+				
+				String[] splitLine = str.split(";");
+				// A very specific regex for matching a datetime string -- Could probably be matched to a specific format that looks cleaner
+				if (splitLine.length > 1 && splitLine[1].matches("( [A-Z][a-z]{2}){2} [0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2} [A-Z]{3} [0-9]{4}")) {
+					String dateTime = ";" + splitLine[1] + "\n";
+					dateTimeCounter++;
+					if (pprjSwitch) {
+						Files.write(Paths.get(pprjFilename), dateTime.getBytes(), StandardOpenOption.APPEND);
+					}
+					if (dateTimeCounter == 2) {
+						Files.write(Paths.get(pontFilename), dateTime.getBytes(), StandardOpenOption.APPEND);
+					}
+					if (dateTimeCounter == 3) {
+						Files.write(Paths.get(pinsFilename), dateTime.getBytes(), StandardOpenOption.APPEND);
+					}
+					continue;
+				}
+				str += "\n";
+				
+				// Generate pprj file
+				if (dateTimeCounter == 1 && pprjSwitch) {
+					if (str.contains("pprj_file_content")) {
+						str = "\n";
+						pprjSwitch = false;
+					}
+					if (str.contains(".pont")) {
+						str = str.replaceAll("[a-zA-Z0-9]+.pont", pontFilename);
+					}
+					if (str.contains(".pins")) {
+						str = str.replaceAll("[a-zA-Z0-9]+.pins", pinsFilename);
+					}
+					Files.write(Paths.get(pprjFilename), str.getBytes(), StandardOpenOption.APPEND);
+				}
+				
+				// Generate pont file
+				if (dateTimeCounter == 2 && str.startsWith(";")) {
+					pontSwitch = true;
+				}
+				if (pontSwitch) {
+					if (str.contains("pont_file_content")) {
+						String[] asciiSplit = str.split("\\)\\)\\)");
+						str = asciiSplit[0] + ")))\n";
+						pontSwitch = false;
+					}
+					Files.write(Paths.get(pontFilename), str.getBytes(), StandardOpenOption.APPEND);
+				}
+				
+				// Generate pins file
+				if (dateTimeCounter == 3) {
+					if (str.contains("pins_file_stub")) {
+						str = "\n";
+					}
+					Files.write(Paths.get(pinsFilename), str.getBytes(), StandardOpenOption.APPEND);
+				}
+			}
+		}
+		pprjWriter.close();
+		pontWriter.close();
+		pinsWriter.close();
+		String outpathName = releaseNumber + "/reactome_data_model.";
+		Files.move(Paths.get(pprjFilename), Paths.get(outpathName + "pprj"), StandardCopyOption.REPLACE_EXISTING); 
+		Files.move(Paths.get(pontFilename), Paths.get(outpathName + "pont"), StandardCopyOption.REPLACE_EXISTING); 
+		Files.move(Paths.get(pinsFilename), Paths.get(outpathName + "pins"), StandardCopyOption.REPLACE_EXISTING); 
```

Comment:
It might be more readable to extract the move implementation into a private method.

```
move(pprjFilename, outpathName + "pprj");
move(pontFilename, outpathName + "pont");
move(pinsFilename, outpathName + "pins");


private static void move(String sourceFile, String targetFile) throws IOException {
    Files.move(Paths.get(sourceFile), Paths.get(targetFile), StandardCopyOption.REPLACE_EXISTING);
}
```

That also gives you flexibility of only needing to edit one place if you ever want to change the implementation for the move.

Reply:
Made the change